File: c++/insertionsort.cpp
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
    int n=5;
    int arr[] = {5,3,1,4,2};
    for(int ele:arr){
      cout<<ele<<" ";
    }
    cout<<endl;

    // insertion sort
    for(int i=1;i<n;i++){
        int j=i;
        while(j>=1 && arr[j-1]>arr[j]){
            swap(arr[j],arr[j-1]);
            j--;
        }
    }
      for(int ele:arr){
         cout<<ele<<" ";
       }
    return 0;

}

File: Sorting_Algorithms/Insertion Sort.py
# Creating Function To Implement Insertion Sort Using Iteration

def insertion_sort(arr):    

    for i in range(1, len(arr)):
        current_element = arr[i]             
        j = i - 1

        while j >= 0 and arr[j] > current_element:
            arr[j + 1] = arr[j]
            j = j - 1

        arr[j + 1] = current_element

    return arr


if __name__ == '__main__':

    List = [32, 3, 2, 1, 112, 12, 31, 21, 42, 32]  # unsorted list

    # Print the original list.
    print("Original List:")
    print(List)

    # Sort the list using the insertion-sort function and print the sorted list.
    sorted_list = insertion_sort(List)
    print("Sorted List:")
    print(sorted_list)
